samples/templates/default-azuredeploy.json

```diff
@@ -327,31 +400,47 @@
             ]
         },
         {
+            "condition": "[equals(parameters('solutionType'),'FhirServerCosmosDB')]",
             "type": "Microsoft.KeyVault/vaults/secrets",
             "name": "[concat(variables('serviceName'), '/CosmosDb--Host')]",
             "apiVersion": "2015-06-01",
             "properties": {
                 "contentType": "text/plain",
-                "value": "[reference(concat('Microsoft.DocumentDb/databaseAccounts/', variables('serviceName'))).documentEndpoint]"
+                "value": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), reference(concat('Microsoft.DocumentDb/databaseAccounts/', variables('serviceName'))).documentEndpoint, '')]"
```

Comment:
>if(equals(parameters('solutionType'),'FhirServerCosmosDB') [](start = 27, length = 58)

nit: not an expert in templates, but would it actually be computed if condition above is false?
If not, can this line can be left as is?

Reply:
Same applies to other changes with condition.

---
In reply to: [516303071](https://github.com/microsoft/fhir-server/pull/1201#discussion_r516303071) [](ancestors = 516303071)

Reply:
The template was failing to deploy without these changes. The reason I found was that references get computed regardless of whether the section they are a part of is used or not. Since that reference was to a service that only gets deployed when Cosmos DB is being used it would fail the SQL deployment with an invalid reference error. Similar for the other changes like this.